concepts.md

```diff
@@ -7,118 +7,118 @@ title: SirixDB - Architecture and Concepts
+: IndirectPages are used to increase the fanout of the tree. SirixDB uses these pages to be able to store and retrieve a large number of records while only ever having to read a predefined number of pages. We currently store 512 references in the `IndirectPage` to either another layer of indirect pages or the data pages, either a `RevisionRootPage` or a `RecordPage`. SirixDB adds a new level of indirect pages whenever it runs out of the number of records it can address in the leaf pages. It stores the height of the current subtree that is the number of levels of indirect pages in the respective subtree-root page. We borrowed the ideas from the filesystem ZFS and hash-array-based tries as we also store checksums in parent database pages/page-fragments, which in turn form a self-validating merkle-tree. As IndirectPages potentially may have many `null`-pointers, SirixDB uses a bitset to keep track of which array indices contain references. Thus, it can store a compact array or list in memory.
 
 RevisionRootPage
-: The `RevisionRootPage` is the main entry point to a revision. It stores the author-ID, an optional commit-message and a timestamp in the form of the unix epoch (milliseconds since 1970). Furthermore it stores a reference to a `PathPage`, a `CASPage` (if it exists), a `NamePage` and an `IndirectPage`. The indirect page is the entry point to the data stored in the leaf `RecordPage`s. The right subtree of the `RevisionRootPage` started by the `IndirectPage` actually is the main entry point to our data stored in the leaf nodes, the `RecordPage`s once again.
-To support fast access to a `RevisionRootPage`, SirixDB stores a second file with just the offsets to specific revisions in a revisions-file,  which is read into main-memory on startup.
+: The `RevisionRootPage` is the main entry point to a revision. It stores the author-ID, an optional commit message, and a timestamp in the form of the unix epoch (milliseconds since 1970). Furthermore, it stores a reference to a `PathPage`, a `CASPage` (if it exists), a `NamePage`, and an `IndirectPage`. The indirect page is the entry point to the data stored in the leaf `RecordPages`. The right subtree of the `RevisionRootPage` started by the `IndirectPage` is the main entry point to our data stored in the leaf nodes, the RecordPages once again. 
```

Comment:
I feel like this paragraph is unclear and could be improved, but I don't understand what is going on here. @JohannesLichtenberger?